api/lib/source/polkadotV0-source.js

```diff
@@ -24,22 +29,61 @@ class polkadotAPI {
     this.db = db
   }
 
+  initialize(config) {
+    this.context = config.context
+    // manually set cache to checking it
+    this.cache = new InMemoryLRUCache()
+    this.httpCache = new HTTPCache(this.cache, this.httpFetch)
+  }
+
   setReducers() {
     this.reducers = require('../reducers/polkadotV0-reducers')
   }
 
+  async getRetry(url, intent = 0) {
+    // check cache size, and flush it if it's bigger than something
+    if ((await this.cache.getTotalSize()) > 100000) {
+      await this.cache.flush()
+    }
+    // clearing memoizedResults
+    this.memoizedResults.clear()
+    try {
+      return await this.get(url, null, { cacheOptions: { ttl: 1 } }) // normally setting cacheOptions should be enought, but...
+    } catch (error) {
+      // give up
+      if (intent >= 3) {
+        console.error(
+          `Error for query ${url} in network ${this.networkId} (tried 3 times)`
+        )
+        throw error
+      }
+
+      // retry
+      await new Promise((resolve) => setTimeout(() => resolve(), 1000))
+      return this.getRetry(url, intent + 1)
+    }
+  }
+
+  // querying data from the sidecar REST API
+  async query(url) {
+    return this.getRetry(url)
+  }
+
   // rpc initialization is async so we always need to assume we need to wait for it to be initialized
   async getAPI() {
     const api = this.store.polkadotRPC
     await api.isReady
     return api
   }
 
-  async getNetworkAccountInfo(address, api) {
+  async getNetworkAccountInfo(address) {
     if (typeof address === `object`) address = address.toHuman()
     if (this.store.identities[address]) return this.store.identities[address]
+    // TODO: We are not handling sub-identities
     const accountInfo = !this.store.validators[address]
-      ? await api.derive.accounts.info(address)
+      ? await this.query(
+          `${this.baseURL}/pallets/identity/storage/identityOf?key1=${address}`
```

Comment:
I don't think you need to specify baseURL if it is specified on the RESTDataSource. See Cosmos